src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBroker.cs

```diff
@@ -71,51 +74,40 @@ private async Task AcquireTokenInternalAsync(IDictionary<string, string> brokerP
-                    _logger.Info("Android Broker - Starting ActionView activity to " + appLink);
-                    _activity.StartActivity(new Intent(Intent.ActionView, AndroidNative.Net.Uri.Parse(appLink)));
-
-                    throw new MsalClientException(
-                        MsalError.BrokerApplicationRequired,
-                        MsalErrorMessage.BrokerApplicationRequired);
-                }
                 await _brokerHelper.InitiateBrokerHandshakeAsync(_activity).ConfigureAwait(false);
 
                 brokerPayload[BrokerParameter.BrokerAccountName] = AndroidBrokerHelper.GetValueFromBrokerPayload(brokerPayload, BrokerParameter.Username);
 
-                // Don't send silent background request if account information is not provided
                 if (brokerPayload.ContainsKey(BrokerParameter.IsSilentBrokerRequest))
                 {
-                    _logger.Verbose("User is specified for silent token request. Starting silent broker request.");
-                    string silentResult = await _brokerHelper.GetBrokerAuthTokenSilentlyAsync(brokerPayload, _activity).ConfigureAwait(false);
-                    if (!string.IsNullOrEmpty(silentResult))
-                    {
-                        s_androidBrokerTokenResponse = MsalTokenResponse.CreateFromAndroidBrokerResponse(silentResult, s_correlationId);
-                    }
-                    else
-                    {
-                        s_androidBrokerTokenResponse = new MsalTokenResponse
-                        {
-                            Error = MsalError.BrokerResponseReturnedError,
-                            ErrorDescription = "Failed to acquire token silently from the broker." + MsalErrorMessage.AndroidBrokerCannotBeInvoked,
-                        };
-                    }
+                    s_androidBrokerTokenResponse = await AcquireTokenSilentViaBrokerAsync(brokerPayload).ConfigureAwait(false);
                     return;
                 }
-                else
-                {
-                    _logger.Verbose("User is not specified for silent token request");
-                }
 
-                _logger.Verbose("Starting Android Broker interactive authentication");
+                await AcquireTokenInteractiveViaBrokerAsync(brokerPayload).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                _logger.ErrorPiiWithPrefix(ex, "Broker invocation failed.");
+                throw;
+            }
+
+            using (_logger.LogBlockDuration("waiting for broker response"))
+            {
+                await s_readyForResponse.WaitAsync().ConfigureAwait(false);
+            }
+        }
 
+        private async Task AcquireTokenInteractiveViaBrokerAsync(IDictionary<string, string> brokerPayload)
+        {
+            using (_logger.LogMethodDuration())
```

Comment:
So instead of doing smth like 

`_logger.LogMethodDuration("AcquireTokenInteractiveViaBrokerAsync")` 

we can just rely on the compiler to fill in the name ....